src/openforms/registrations/contrib/stuf_zds/tests/test_backend.py

```diff
@@ -1211,3 +1234,320 @@ def test_reference_can_be_extracted(self, m):
         reference = extract_submission_reference(submission)
 
         self.assertEqual("abcd1234", reference)
+
+
+@tag("gh-1183")
+@freeze_time("2020-12-22")
+@temp_private_root()
+class PartialRegistrationFailureTests(VariablesTestMixin, StUFZDSTestBase):
+    """
+    Test that partial results are stored and don't cause excessive registration calls.
+
+    Issue #1183 -- case numbers are reserved to often, as a retry reserves a new one. It
+    also happens that when certain other calls fail, a new Zaak is created which
+    should not have been created again.
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.service = StufServiceFactory.create()
+        config = StufZDSConfig.get_solo()
+        config.service = cls.service
+        config.save()
```

Comment:
Should this be a mock?

Reply:
If I remember correctly, the config is loaded in multiple places so that would require patching in multiple places. This would then be simpler.